test/v1/controllers/RetrieveTransactionDetailsControllerSpec.scala

```diff
@@ -0,0 +1,196 @@
+    MockedEnrolmentsAuthService.authoriseUser()
+  }
+
+  "retrieveTransactionDetails" should {
+
+    "return a valid transactions response" when {
+      "a request sent has valid details" in new Test {
+
+        MockRetrieveTransactionDetailsRequestParser
+          .parse(rawRequest)
+          .returns(Right(parsedRequest))
+
+        MockRetrieveTransactionDetailsService
+          .retrieveTransactionDetails(parsedRequest)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, mtdResponse))))
+
+        MockHateoasFactory
+          .wrap(mtdResponse, RetrieveTransactionDetailsHateoasData(nino, transactionId, Some(paymentId)))
+          .returns(HateoasWrapper(hateoasResponse,
+            Seq(transactionsHateoasLink, paymentAllocationHateoasLink)))
+
+        val result: Future[Result] = controller.retrieveTransactionDetails(nino, transactionId)(fakeGetRequest)
+
+        status(result) shouldBe OK
+        contentAsJson(result) shouldBe mtdJson
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+      }
+    }
+    "return the correct errors" when {
+      "parser errors occur" must {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockRetrieveTransactionDetailsRequestParser
+              .parse(rawRequest)
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.retrieveTransactionDetails(nino, transactionId)(fakeGetRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (TransactionIdFormatError, BAD_REQUEST)
+        )
+
+        input.foreach(args => (errorsFromParserTester _).tupled(args))
+      }
+    }
+
+    "service errors occur" must {
+      def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
+        s"a $mtdError error is returned from the service" in new Test {
+
+          MockRetrieveTransactionDetailsRequestParser
+            .parse(rawRequest)
+            .returns(Right(parsedRequest))
+
+          MockRetrieveTransactionDetailsService
+            .retrieveTransactionDetails(parsedRequest)
+            .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), mtdError))))
+
+          val result: Future[Result] = controller.retrieveTransactionDetails(nino, transactionId)(fakeGetRequest)
+
+          status(result) shouldBe expectedStatus
+          contentAsJson(result) shouldBe Json.toJson(mtdError)
+          header("X-CorrelationId", result) shouldBe Some(correlationId)
+        }
+      }
+
+      val input = Seq(
```

Comment:
can you add (NinoFormatError, BAD_REQUEST),
        (TransactionIdFormatError, BAD_REQUEST),

Reply:
Those tests are already there, in the above test "parser errors occur"

Reply:
True, but service layer also throwing the same errors... that the reason I asked to add